slither/tools/read_storage/__main__.py

```diff
@@ -126,22 +126,21 @@ def main() -> None:
     else:
         contracts = slither.contracts
 
-    srs = SlitherReadStorage(contracts, args.max_depth)
-
-    try:
-        srs.block = int(args.block)
-    except ValueError:
-        srs.block = str(args.block or "latest")
-
+    rpc_info = None
     if args.rpc_url:
-        # Remove target prefix e.g. rinkeby:0x0 -> 0x0.
-        address = target[target.find(":") + 1 :]
-        # Default to implementation address unless a storage address is given.
-        if not args.storage_address:
-            args.storage_address = address
-        srs.storage_address = args.storage_address
-
-        srs.rpc = args.rpc_url
+        try:
+            block = int(args.block)
+        except ValueError:
+            block = BlockTag(args.block) or "latest"
```

Comment:
```suggestion
            block = BlockTag(args.block)
```
"latest" is the default arg and not being converted to `BlockTag`